code/engine/xrCore/_math.cpp

```diff
@@ -201,90 +201,16 @@ void _initialize_cpu() {
     Didentity.identity();  // Identity matrix
     pvInitializeStatics(); // Lookup table for compressed normals
     FPU::initialize();
-    _initialize_cpu_thread();
-
-    g_initialize_cpu_called = true;
-}
-
-// per-thread initialization
-#include <xmmintrin.h>
-#define _MM_DENORMALS_ZERO_MASK 0x0040
-#define _MM_DENORMALS_ZERO_ON 0x0040
-#define _MM_FLUSH_ZERO_MASK 0x8000
-#define _MM_FLUSH_ZERO_ON 0x8000
-#define _MM_SET_FLUSH_ZERO_MODE(mode) _mm_setcsr((_mm_getcsr() & ~_MM_FLUSH_ZERO_MASK) | (mode))
-#define _MM_SET_DENORMALS_ZERO_MODE(mode) \
-    _mm_setcsr((_mm_getcsr() & ~_MM_DENORMALS_ZERO_MASK) | (mode))
-static BOOL _denormals_are_zero_supported = TRUE;
-extern void __cdecl _terminate();
-void debug_on_thread_spawn();
-
-void _initialize_cpu_thread() {
-    debug_on_thread_spawn();
 #ifndef XRCORE_STATIC
     // fpu & sse
     FPU::m24r();
 #endif // XRCORE_STATIC
-    if (CPU::ID.hasFeature(CpuFeature::Sse)) {
-        //_mm_setcsr ( _mm_getcsr() | (_MM_FLUSH_ZERO_ON+_MM_DENORMALS_ZERO_ON) );
-        _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
-        if (_denormals_are_zero_supported) {
-            __try {
-                _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
-            } __except (EXCEPTION_EXECUTE_HANDLER) {
-                _denormals_are_zero_supported = FALSE;
-            }
-        }
-    }
 }
 
-// threading API
-#pragma pack(push, 8)
-struct THREAD_NAME {
-    DWORD dwType;
-    LPCSTR szName;
-    DWORD dwThreadID;
-    DWORD dwFlags;
-};
-void thread_name(const char* name) {
-    THREAD_NAME tn;
-    tn.dwType = 0x1000;
-    tn.szName = name;
-    tn.dwThreadID = DWORD(-1);
-    tn.dwFlags = 0;
-    __try {
-        RaiseException(0x406D1388, 0, sizeof(tn) / sizeof(DWORD), (ULONG_PTR*)&tn);
-    } __except (EXCEPTION_CONTINUE_EXECUTION) {
-    }
-}
-#pragma pack(pop)
-
-struct THREAD_STARTUP {
-    thread_t* entry;
-    char* name;
-    void* args;
-};
-void __cdecl thread_entry(void* _params) {
-    // initialize
-    THREAD_STARTUP* startup = (THREAD_STARTUP*)_params;
-    thread_name(startup->name);
-    thread_t* entry = startup->entry;
-    void* arglist = startup->args;
-    xr_delete(startup);
-    _initialize_cpu_thread();
-
-    // call
-    entry(arglist);
-}
-
-void thread_spawn(thread_t* entry, const char* name, unsigned stack, void* arglist) {
-    Debug._initialize();
-
-    THREAD_STARTUP* startup = xr_new<THREAD_STARTUP>();
-    startup->entry = entry;
-    startup->name = (char*)name;
-    startup->args = arglist;
-    _beginthread(thread_entry, stack, startup);
+void set_thread_name(std::wstring_view name)
+{
+    if (IsWindows10OrGreater)
+        SetThreadDescription(GetCurrentThread(), name.data());
```

Comment:
string_view::data is not guaranteed to be a null terminated, just use const wchar_t*